test/decl/var/property_wrappers.swift

```diff
@@ -877,3 +877,38 @@ struct TestComposition {
     $p3 = d // expected-error{{cannot assign value of type 'Double' to type 'WrapperD<WrapperE<Int?>, Int, String>'}}
 	}
 }
+
+// ---------------------------------------------------------------------------
+// Missing Property Wrapper Unwrap Diagnostics
+// ---------------------------------------------------------------------------
+@propertyWrapper
+struct Foo<T> {
+  var wrappedValue: T
+
+  func foo() {}
+}
+
+@propertyWrapper
+struct Bar<T, V> {
+  var wrappedValue: T
+
+  func bar() {}
+}
+
+extension Bar where V == String { // expected-note {{where 'V' = 'Bool'}}
+  func barWhereVIsString() {}
+}
+
+struct MissingPropertyWrapperUnwrap {
+  @Foo var x: Int
+  @Bar<Int, Bool> var y: Int
+  @Bar<Int, String> var z: Int
+
+  func baz() {
+    self.x.foo() // expected-error {{property 'x' will be unwrapped to value of type 'Int', use '$' to refer to wrapper type 'Foo<Int>'}}{{10-10=$}}
```

Comment:
This is great stuff! The wording here feels a little implementation-oriented to me, though. When I see `@Foo var x: Int`, I'm not really thinking of it being wrapped and unwrapped, I'm thinking that `x` is an `Int` that's also a `Foo`. Maybe something like this?

{{property 'x' of 'Int' has no member named 'foo()'; did you mean '$x' to access the member on 'Foo<Int>'?}}

Reply:
I agree, we have discussed it in one of the earlier comments. I think it should move closer to what we have for conditional conformances where it would talk about reference to member instead of wrapper/unwrap thing.